File: slizaa-hierarchicalgraph/graphdb/org.slizaa.hierarchicalgraph.graphdb.mapping.service/src/main/java/org/slizaa/hierarchicalgraph/graphdb/mapping/service/IMappingService.java
package org.slizaa.hierarchicalgraph.graphdb.mapping.service;

import org.slizaa.core.boltclient.IBoltClient;
import org.slizaa.hierarchicalgraph.core.model.HGRootNode;
import org.slizaa.hierarchicalgraph.graphdb.mapping.service.internal.DefaultMappingService;
import org.slizaa.hierarchicalgraph.graphdb.mapping.spi.IMappingProvider;

public interface IMappingService {

  /**
   * <p>
   * </p>
   *
   * @param mappingProvider
   * @param boltClient
   *
   * @return
   * @throws MappingException
   */
  HGRootNode convert(IMappingProvider mappingProvider, IBoltClient boltClient)
      throws MappingException;

    public static IMappingService createHierarchicalgraphMappingService() {
    return new DefaultMappingService();
  }
}

File: slizaa-hierarchicalgraph/graphdb/io.hierograph.hierarchicalgraph.graphdb.mapping.service/src/main/kotlin/io/hierograph/hierarchicalgraph/graphdb/mapping/service/IMappingService.kt
package io.hierograph.hierarchicalgraph.graphdb.mapping.service

import io.hierograph.hierarchicalgraph.core.model.HGRootNode
import io.hierograph.hierarchicalgraph.graphdb.mapping.spi.IMappingProvider
import org.slizaa.core.boltclient.IBoltClient

interface IMappingService {
    fun convert(mappingProvider: IMappingProvider, boltClient: IBoltClient): HGRootNode
}
